KIPs/kip-81.md

```diff
@@ -0,0 +1,773 @@
+  - **CnStaking admins**: A set of accounts controlled by each GC which can manage the staked KLAYs and its voter account. Every CnStakingV2 contract can have different set of admins.
+  - **Voter account**: An account controlled by each GC which can cast on-chain votes. But this account cannot withdraw KLAYs from CnStakingV2 contracts.
+  - **Secretariat account**: An account controlled by the Foundation which can propose and execute on-chain governance proposals.
+
+![contracts and accounts](../assets/kip-81/smart_contract_relation.png)
+
+### AdressBook
+
+In Cypress mainnet and Baobab testnet, an [AddressBook contract](https://github.com/klaytn/klaytn/blob/v1.9.1/contracts/reward/contract/AddressBook.sol) is deployed at address `0x0000000000000000000000000000000000000400`. For the purpose of Voting, following function of the AddressBook is used.
+
+```solidity
+interface IAddressBook {
+    /// @dev Returns all addresses in the AddressBook.
+    ///
+    /// Each GC's addresses are stored in the same index of the three arrays.
+    /// i.e. GC[i] is described in (nodeIds[i], stakingContracts[i], rewardAddresses[i])
+    /// However, a GC may operate multiple staking contracts. In this case,
+    /// the GC occupies multiple indices with the same reward address.
+    ///
+    /// @return nodeIds           GC consensus node address list
+    /// @return stakingContracts  GC staking contract address list
+    /// @return rewardAddresses   GC reward recipient address list
+    /// @return pocAddress        PoC(KGF) recipient address
+    /// @return kirAddress        KIR recepient address
+    function getAllAddressInfo() external view returns(
+        address[] memory nodeIds,
+        address[] memory stakingContracts,
+        address[] memory rewardAddresses,
+        address pocAddress,
+        address kirAddress);
+}
+```
+
+### CnStakingV2
+
+CnStakingV2 is an upgraded version of CnStakingContract. The CnStakingContract has been serving the purpose of locking GC stakes. V2 will add two new features related to StakingTracker, which will be described later.
+
+- Whenever its KLAY balance changes, V2 notifies the StakingTracker.
+- V2 stores a voter account address, and notifies the StakingTracker whenever it changes. The voter account can be changed by its admins.
+
+To support the new features, following functions are added.
+
+```solidity
+abstract contract CnStakingV2 {
+    event UpdateStakingTracker(address stakingTracker);
+    event UpdateVoterAddress(address voterAddress);
+
+    IStakingTracker public stakingTracker;
+    address public voterAddress;
+
+    /// @dev Set the initial stakingTracker address
+    /// Emits an UpdateStakingTracker event.
+    /// This function only works before contract initialization
+    function setStakingTracker(address _tracker) external beforeInit;
+
+    /// @dev Update the StakingTracker address this contract points to
```

Comment:
@blukat29 It seems like `updateStakingTracker` function also emits `UpdateStakingTracker`. Is this right?

Reply:
Updated f450b5b.